### PR TITLE
Add an ilog2 function.

### LINKED ISSUE #167: add an ilog2 function
We frequently work with powers of 2, in particular, matrices whose dimensions are 2^n x 2^n. Sometimes we need to compute n.

The way we compute `n` for `x = 2^n` is `(1- (integer-length x))` when `x` is a known power of two.

First, write a function called `ilog2` which does this computation, and second, replace all instances of `1- integer-length` with this function to make the code clearer.

### PR DESCRIPTION
**Pull request by fengdlm**
Closes #167. An ilog2 function has been added and all the calls to `(1- (integer-length x)` has been replace.

### CODE REVIEW

File: src/matrix-operations.lisp [hunk 1]
```diff
@@ -26,6 +26,10 @@
         :always (>= (+ x +double-comparison-threshold-strict+)
                     (- y +double-comparison-threshold-strict+))))
 
+(defun ilog2 (x)
```
Comment (stylewarning) [inline src/matrix-operations.lisp]: I suggest putting this in `utilities.lisp` and exporting it from `package.lisp`.

File: src/matrix-operations.lisp [hunk 2]
```diff
@@ -26,6 +26,10 @@
...
+  "Compute integer logarithm of x to the base 2."
```
Comment (stylewarning) [inline src/matrix-operations.lisp]: Capitalize `X`.

File: src/clifford/benchmarking-procedures.lisp [hunk 3]
```diff
@@ -171,7 +171,7 @@
   (loop :for gate-application :across (quil::parsed-program-executable-code parsed-quil)
         :collect
         (let* ((gate (quil:gate-matrix gate-application))
-               (num-qubits (1- (integer-length (magicl:matrix-cols gate)))))
+               (num-qubits (quil::ilog2 (magicl:matrix-cols gate))))
```
Comment (stylewarning) [inline src/clifford/benchmarking-procedures.lisp]: See comment elsewhere about exporting this, so as to write `quil:ilog2`.

**Comment by fengdlm** [comment]
Hi @notmgsk @stylewarning 
Thanks for reviewing the code! I made the changes as suggested. Please let me know if you think more changes are necessary. Thanks!

**Comment by stylewarning** [comment]
Thanks!

**Comment by stylewarning** [comment]
Keep em coming. :)

**Pull request merged**